### PR TITLE
[P4Orch] Lazy UDF match creation to avoid failure during warm reboot

### PR DESCRIPTION
**Pull request by baxia-lan**
Signed-off-by: Yilan Ji [yilanji@google.com](mailto:yilanji@google.com)

**What I did**
Lazy UDF match creation, create once the first ACL table with UDF is created in P4Orch

**Why I did it**
Temp solution for [issue](https://github.com/Azure/sonic-buildimage/issues/10410).  

**How I verified it**

**Details if related**
Warm reboot in P4Orch needs to be addressed as the intent view is applied not the programmed view.

### CODE REVIEW

**Review by prsunny — DISMISSED** [review]

**Comment by prsunny** [comment]
Please add VS coverage

File: orchagent/p4orch/acl_table_manager.cpp [hunk 1]
```diff
@@ -465,8 +464,13 @@ ReturnCode AclTableManager::createUdf(const P4UdfField &udf_field)
     sai_object_id_t udf_match_oid;
     if (!m_p4OidMapper->getOID(SAI_OBJECT_TYPE_UDF_MATCH, P4_UDF_MATCH_DEFAULT, &udf_match_oid))
     {
-        return ReturnCode(StatusCode::SWSS_RC_NOT_FOUND)
-               << "UDF default match " << QuotedVar(P4_UDF_MATCH_DEFAULT) << " does not exist";
+        // Create the default UDF match
+        auto status = createDefaultUdfMatch();
+        if (!status.ok())
+        {
+            SWSS_LOG_ERROR("Failed to create ACL UDF default match : %s", status.message().c_str());
```
Comment (mint570) [inline orchagent/p4orch/acl_table_manager.cpp:471]: Return error?

**Review by mint570 — COMMENTED** [review]
Need to update p4orch unit test and pytest.
For p4orch unit test, currently it is broken. We will need https://github.com/Azure/sonic-swss/pull/2237 to be submitted (Even in pr2237, it might be broken since I did a rebase. But it should make things a lot easier if we have pr2237.).

**Comment by baxia-lan** [comment]
The tests/p4rt/test_p4rt_acl.py should cover the UDF match creation code, but it seems not been counted?

**Comment by baxia-lan** [comment]
The tests/p4rt/test_p4rt_acl.py should cover the UDF match creation code, but it seems not been counted?

**Comment by baxia-lan** [comment]
@prsunny  Kindly ping as a reminder. Thanks a lot 🙏

File: orchagent/p4orch/acl_table_manager.cpp [hunk 2]
```diff
@@ -36,16 +36,15 @@ AclTableManager::AclTableManager(P4OidMapper *p4oidMapper, ResponsePublisherInte
     SWSS_LOG_ENTER();
 
     assert(p4oidMapper != nullptr);
-    // Create the default UDF match
-    auto status = createDefaultUdfMatch();
```
Comment (prsunny) [inline orchagent/p4orch/acl_table_manager.cpp:40]: Where is it created now?

**Comment by prsunny** [comment]
Diff coverage failure acceptable as the code added is in C++ destructor

image

File: orchagent/p4orch/acl_table_manager.cpp (hunk 2 above)
Comment (baxia-lan) [reply orchagent/p4orch/acl_table_manager.cpp:40]: Line#468, when the ACL table with UDF fields defined created for the first time by p4orch.

**Review by prsunny — APPROVED** [review]

**Pull request merged**